packages/core-mobile/app/screens/send/SendTokenScreen.tsx

```diff
@@ -0,0 +1,146 @@
+import { SendTokensScreenProps } from 'navigation/types'
+import AppNavigation from 'navigation/AppNavigation'
+import { useNavigation, useRoute } from '@react-navigation/native'
+import AnalyticsService from 'services/analytics/AnalyticsService'
+import { audioFeedback, Audios } from 'utils/AudioFeedback'
+import { isUserRejectedError } from 'store/rpc/providers/walletConnect/utils'
+import { showTransactionErrorToast } from 'utils/toast'
+import { getJsonRpcErrorMessage } from 'utils/getJsonRpcErrorMessage'
+import { selectTokensWithBalance } from 'store/balance'
+import { useSelector } from 'react-redux'
+import {
+  NetworkTokenWithBalance,
+  TokenType,
+  TokenWithBalanceAVM,
+  TokenWithBalanceBTC,
+  TokenWithBalancePVM
+} from '@avalabs/vm-module-types'
+import { selectActiveAccount } from 'store/account'
+import SendEVM from './components/SendEVM'
+import SendBTC from './components/SendBTC'
+import SendAVM from './components/SendAVM'
+import SendPVM from './components/SendPVM'
+
+type SendTokenScreenNavigationProp = SendTokensScreenProps<
+  typeof AppNavigation.Send.Send
+>
+
+const SendTokenScreen: FC = () => {
+  const navigation =
+    useNavigation<SendTokenScreenNavigationProp['navigation']>()
+  const { params } = useRoute<SendTokenScreenNavigationProp['route']>()
+  const contact = params?.contact
+
+  const { setToken, setToAccount } = useSendContext()
+  const { activeNetwork } = useNetworks()
+  const activeAccount = useSelector(selectActiveAccount)
+  const tokens = useSelector(selectTokensWithBalance)
+  const nativeToken = tokens.find(t => t.type === TokenType.NATIVE)
+
+  useEffect(() => {
+    if (params?.token) {
+      setToken(params?.token)
+    }
+  }, [setToken, params?.token])
+
+  useEffect(() => {
+    if (contact) {
+      setToAccount({
+        address: contact.address,
+        title: contact.name
+      })
+    }
+  }, [contact, setToAccount])
+
+  const handleOpenAddressBook = (): void => {
+    navigation.navigate(AppNavigation.Wallet.AddressBook)
+  }
+
+  const handleOpenQRScanner = (): void => {
+    navigation.navigate(AppNavigation.Modal.QRScanner, {
+      onSuccess: (data: string) => {
+        setToAccount({ address: data, title: '' })
+      }
+    })
+  }
+
+  const handleSuccess = (txHash: string): void => {
+    AnalyticsService.captureWithEncryption('SendTransactionSucceeded', {
+      chainId: activeNetwork.chainId,
+      txHash
+    })
+    audioFeedback(Audios.Send)
+
+    navigation.goBack()
+  }
+
+  const handleFailure = (error: unknown): void => {
+    if (error instanceof Error && !isUserRejectedError(error)) {
+      showTransactionErrorToast({
+        message: getJsonRpcErrorMessage(error)
+      })
+      AnalyticsService.capture('SendTransactionFailed', {
+        errorMessage: error.message,
+        chainId: activeNetwork.chainId
+      })
+    }
+  }
+
+  const isLoading = !activeAccount || !nativeToken
+
+  if (isLoading) {
+    return <></>
+  }
+
```

Comment:
we should probably render something when network is none of the supported ones

Reply:
added `<ActivityIndicator />` for loading and `<ZeroState />` for non-supported network